### ISSUE TITLE
Specify specs and readonly configuration (chromeWebSecurity) per matrix container

### ISSUE THREAD

**Issue by a8trejo** [cypress-io/github-action#1137]
Hello everyone!

I'm encountering a situation in which I need a specific spec file "**spec1**" to run with `chromeWebSecurity` as `true`, and the rest of the suite with `false` due to iframes.

So far the only solution/workaround I've found is to create a specific GHA job for **spec1** with `chromeWebSecurity` as `true`, and passing the respective spec pattern to`spec` (no matrix strategy here since it's 1 spec)

And having a diff job (with matrix strategy) for the rest of the suite with the respective `excludeSpecPattern` and `chromeWebSecurity` as `false`

This causes duplicate entries in Cypress Cloud since Cypress GHA won't let me use the same `ci-build-id`, when I tried using the same ID for both jobs I got the error `You passed the --parallel flag, but we do not parallelize tests across different environments` .

In the meantime I'm using a `tag` to be able to tell from one another, when in reality they are both the same test run.

Wanted to flag this as a nice enhancement to the action, the possibility to specify configurations per spec or/and container?

Thank you very much!

**Comment by MikeMcC399** [comment]
@a8trejo

I understand that you would like the Cypress GitHub Action to allow you to either specify different [configuration options](https://docs.cypress.io/guides/references/configuration) per Cypress spec or per GitHub Action container. At a first reading, it does not look like this would be a possibility for the action.

## Per spec

The [Configuration > Overriding Options](https://docs.cypress.io/guides/references/configuration#Overriding-Options) describes the case of dynamically altering configuration options. The [Browser option](https://docs.cypress.io/guides/references/configuration#Browser) `chromeWebSecurity` you specifically mention is however not on the list [Test configuration](https://docs.cypress.io/guides/references/configuration#Test-Configuration) of options which can be changed at run time. I think you are aware of this, since you used the term "readonly" in the subject of your request.

## Per matrix container

The allocation of test specs to matrix containers, running tests in parallel in the Cypress Cloud, is controlled by the Cloud software, not by the Cypress GitHub Action. This is part of the [Cypress Cloud > Smart Orchestration > Load Balancing](https://docs.cypress.io/guides/cloud/smart-orchestration/load-balancing) strategy.

## Suggestion

I would highly recommend the Cypress technical community on Discord as a resource for you to use
[![Discord chat](https://img.shields.io/badge/chat-on%20Discord-brightgreen)](https://on.cypress.io/discord) (click on button). There is a specific channel `cypress-cloud` where you can get advice.

**Comment by a8trejo** [comment]
Thank you @MikeMcC399 !, I do understand `chromeWebSecurity` is readonly as a dynamic configuration, hence my workaround of having a specific Github actions job only for the specs that need it. 

Is there a place to submit feature requests to the Cypress Cloud team? Should it be with our cypress account executive?

**Comment by MikeMcC399** [comment]
@a8trejo 

I can't say what the best way is to engage with the Cypress Cloud team as I am an external contributor and I'm not part of the internal Cypress.io team.

If you have an Account Executive allocated to you they would be able to advise you I imagine. I have also seen feature requests submitted through Discord picked up. https://www.cypress.io/support includes some information about submitting support tickets for Cypress Cloud.

**Comment by MikeMcC399** [comment]
@a8trejo 

I am going to close this request as I don't see any technical scope at this time for implementing the request in the action. If there is any new information to change this opinion we can re-open the issue.
